### ISSUE TITLE
[Bug] [iOS] Brush color lost on swiping

### ISSUE THREAD

**Issue by tompi** [xamarin/Xamarin.Forms#13376]
### Description

Brush-bindings seems to get lost when navigating in carousel view(only happens on iOS, android works fine).
Happens on both emulator and devices

### Steps to Reproduce

1. Download repro-app: https://github.com/tompi/shapedebug
2. Run it on ios and swipe a few times back and forth

### Expected Behavior

Stroke and fill attributes of shapes are intact after swiping

### Actual Behavior

On third right-swipe, stroke and fill seem to be reset...

### Basic Information

- Version with issue: 5.0.0.1874
- Last known good version: Not tested on prior versions
- Platform Target Frameworks: 
  - iOS:  14.2
  - Android: 10.0

- Affected Devices: iphone 12, ipad pro 12.9", iphone 12 emulator

### Environment

<details><summary>Show/Hide Visual Studio info</summary>

```
=== Visual Studio Professional 2019 for Mac ===

Version 8.8.4 (build 30)
Installation UUID: af35bc1a-d926-4898-808a-505e675fb8b9
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 6.18.0.23 (d16-6 / 088c73638)

	Package version: 612000113

=== Mono Framework MDK ===

Runtime:
	Mono 6.12.0.113 (2020-02/4fdfb5b1fd5) (64-bit)
	Package version: 612000113

=== Roslyn (Language Service) ===

3.8.0-5.20519.18+4c195c3ac1974edcefa76774d7a59a2350ec55fa

=== NuGet ===

Version: 5.8.0.6860

=== .NET Core SDK ===

SDK: /usr/local/share/dotnet/sdk/5.0.101/Sdks
SDK Versions:
	5.0.101
	5.0.100
	3.1.404
	3.1.403
	3.1.402
	3.1.401
	3.1.302
	3.1.301
	3.1.300
	3.1.201
	3.1.200
	3.1.102
	3.1.101
	3.1.100
	3.0.101
	3.0.100
	2.2.207
	2.2.203
	2.1.505
	2.1.504
MSBuild SDKs: /Applications/Visual Studio.app/Contents/Resources/lib/monodevelop/bin/MSBuild/Current/bin/Sdks

=== .NET Core Runtime ===

Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	5.0.1
	5.0.0
	3.1.10
	3.1.9
	3.1.8
	3.1.7
	3.1.6
	3.1.5
	3.1.4
	3.1.3
	3.1.2
	3.1.1
	3.1.0
	3.0.1
	3.0.0
	2.2.8
	2.2.4
	2.1.23
	2.1.22
	2.1.21
	2.1.20
	2.1.19
	2.1.18
	2.1.17
	2.1.16
	2.1.15
	2.1.14
	2.1.13
	2.1.9
	2.1.8

=== .NET Core 3.1 SDK ===

SDK: 3.1.404

=== Xamarin.Profiler ===

Version: 1.6.12.26
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

=== Updater ===

Version: 11

=== Xamarin.Android ===

Version: 11.1.0.17 (Visual Studio Professional)
Commit: xamarin-android/d16-8/c0e2b8e
Android SDK: /Users/tompi/Library/Developer/Xamarin/android-sdk-macosx
	Supported Android versions:
		7.1 (API level 25)
		8.0 (API level 26)
		8.1 (API level 27)

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 30.0.5
SDK Build Tools Version: 29.0.2

Build Information: 
Mono: be2226b
Java.Interop: xamarin/java.interop/d16-8@79d9533
ProGuard: Guardsquare/proguard/proguard6.2.2@ebe9000
SQLite: xamarin/sqlite/3.32.1@1a3276b
Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-8@2fb1cbc

=== Microsoft OpenJDK for Mobile ===

Java SDK: /Users/tompi/Library/Developer/Xamarin/jdk/microsoft_dist_openjdk_8.0.25
1.8.0-25
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

=== Android SDK Manager ===

Version: 16.8.0.32
Hash: 01a7774
Branch: remotes/origin/d16-8
Build date: 2020-12-10 00:32:24 UTC

=== Android Device Manager ===

Version: 16.8.0.46
Hash: 0a81419
Branch: remotes/origin/d16-8
Build date: 2020-12-10 00:32:44 UTC

=== Apple Developer Tools ===

Xcode 12.3 (17715)
Build 12C33

=== Xamarin.Mac ===

Xamarin.Mac not installed. Can't find /Library/Frameworks/Xamarin.Mac.framework/Versions/Current/Version.

=== Xamarin.iOS ===

Version: 14.6.0.15 (Visual Studio Professional)
Hash: 87a1b18d8
Branch: d16-8
Build date: 2020-11-16 21:39:41-0500

=== Xamarin Designer ===

Version: 16.8.0.510
Hash: 44e3f3ce9
Branch: remotes/origin/d16-8
Build date: 2020-12-10 00:06:14 UTC

=== Build Information ===

Release ID: 808040030
Git revision: eaa21a8983aa5c4bcf6a665892478b4f3d413418
Build date: 2020-12-10 11:06:12-05
Build branch: release-8.8
Xamarin extensions: eaa21a8983aa5c4bcf6a665892478b4f3d413418

=== Operating System ===

Mac OS X 10.16.0
Darwin 20.2.0 Darwin Kernel Version 20.2.0
    Wed Dec  2 20:39:59 PST 2020
    root:xnu-7195.60.75~1/RELEASE_X86_64 x86_64


```

</details>

### Screenshots

Screenshots before and after swiping:
![Screenshot 2021-01-12 at 13 00 09](https://user-images.githubusercontent.com/231717/104311911-2b3bb580-54d6-11eb-8c97-3702b5deec7a.png)
![Screenshot 2021-01-12 at 12 59 57](https://user-images.githubusercontent.com/231717/104311917-2c6ce280-54d6-11eb-9aaa-6f9066841880.png)

### Reproduction Link

https://github.com/tompi/shapedebug

### Workaround

This only happens when binding a contentviews "content", does not happen in a "plain" layout...

**Comment by jsuarezruiz** [comment]
Can reproduce the issue. Thanks for the attached sample!.

**Comment by jsuarezruiz** [comment]
@tompi The issue will be fixed by #13386

**Comment by tompi** [comment]
@jsuarezruiz Nice, that was fixed quickly! thnx 👍

**Comment by SeRgI1982** [comment]
It happens not only with CarouselView.
I have experienced it when I used Shapes with TabView. 
I had Shapes on 1st Tab. Switching to another Tab and back cause loss of color (IsSwipeEnabled = false).

Another case: Polygon with Fill color used on my custom control. Changing visibility of this control cause loss of color.
I tried with binded color via StaticResource like also with fixed color (predefined or hex). 
Shapes loss color (Rectangle, Path, Polygon). Line shape works good
